pydantic/types.py

```diff
@@ -2580,7 +2593,7 @@ def decode_str(self, data: bytes, _: core_schema.ValidationInfo) -> str:
         Returns:
             The decoded data.
         """
-        return data.decode()
+        return self.encoder.decode(data.encode()).decode()
```

Comment:
So that I understand the intent here - we validate a `str`, then `encode` -> `bytes`, `decode` with our `encoder`, then `decode` back to a `str`? 🤮

Reply:
yeah, since the encoder operates on bytes I think you need to do this if you want to apply it to strings